contracts/src/BackedCommunityTokenV1.sol

```diff
@@ -118,12 +124,22 @@ contract BackedCommunityTokenV1 is
             descriptor.tokenURI(
                 owner,
                 scores,
-                accessoryIdToArtContract[addressToAccessoryEnabled[owner]]
+                accessoryIdToAccessory[addressToAccessoryEnabled[owner]]
+                    .artContract
             );
     }
 
     // === internal & private ===
     function _unlockAccessory(AccessoryUnlockChange memory change) internal {
+        IBackedCommunityTokenV1.Accessory
+            memory accessory = accessoryIdToAccessory[change.accessoryId];
+        if (accessory.xpBased) {
```

Comment:
don't you need an else check to see if they have been nominated for this one?

Reply:
well my thinking was since this is an admin only method, we can trust that they were nominated / approved if they got all the way to the contract

Reply:
ah, I was thinking that would have
- some `setTrait` method where the user can select which trait to where. 
  - if trait is xp based, check if they qualify xp
  - if trait is not xp based, look up some mapping where admin can set if they can access that trait. 
- view function to get all traits accessible to a user, loop through all traits for the user and do these checks and return an array of ids or strings

Reply:
so then also some admin `awaitTrait` method to set that mapping for non-xp traits?

Reply:
ah i see what you mean! makes sense, i was thinking along similar lines, but with what i have implemented now, there is an extra step to get xp based traits, which your proposition bypasses. will refactor

Reply:
@wilsoncusack updated to something more similar to what you outlined